dask/bytes/core.py

```diff
@@ -141,7 +143,7 @@ def write_bytes(data, urlpath, name_function=None, compression=None,
         raise NotImplementedError("Unknown protocol for writing %s (%s)" %
                                   (protocol, urlpath))
 
-    files = open_files_write(paths, **storage_options)
+    files = [(open_files_write, p, storage_options) for p in paths]
     out = [delayed(write_block_to_file)(v, f, compression, encoding)
            for (v, f) in zip(data, files)]
```

Comment:
I have tried your code, and I find that "data" that gets passed to write_block_to_file is a delayed object (but the file is opened correctly). 
Computing that delayed object does get the right result, but a delayed function should not be computing within itself.